app/controllers/bulk_edit_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+class BulkEditController < ApplicationController
+  include Blacklight::SearchHelper
+
+  def resources_edit
+    (solr_response, _document_list) = search_results(q: params["q"], f: params["f"])
+    @resources_count = solr_response["response"]["numFound"]
+  end
+
+  def resources_update
+    args = {}.tap do |hash|
+      hash[:mark_complete] = (params["mark_complete"] == "1")
+    end
+    batches = prepare_batches
```

Comment:
Can you just rename `prepare_batches` to `batches` and memoize?

Reply:
👍